### PR TITLE
Ensure that tmpfs mounts do not have symlinks

### PR DESCRIPTION
**Pull request by mheon**
When mounting a tmpfs, runc attempts to make the directory it will be mounted at. Unfortunately, Golang's os.MkdirAll deals very poorly with symlinks being part of the path. I looked into fixing this in runc, but it's honestly much easier to just ensure we don't trigger the issue on our end.

Fixes BZ #1686610

### CODE REVIEW

**Comment by rhatdan** [comment]
LGTM

File: libpod/container_internal_linux.go [hunk 1]
```diff
@@ -366,6 +367,18 @@ func (c *Container) generateSpec(ctx context.Context) (*spec.Spec, error) {
...
 	rootPropagation := ""
 	for _, m := range mounts {
+		// We need to remove all symlinks from tmpfs mounts.
+		// Runc and other runtimes may choke on them.
+		// Easy solution: use securejoin to do a scoped evaluation of
+		// the links, then trim off the mount prefix.
+		if m.Type == "tmpfs" {
+			finalPath, err := securejoin.SecureJoin(c.state.Mountpoint, m.Destination)
+			if err != nil {
+				return nil, errors.Wrapf(err, "error resolving symlinks for mount destination %s", m.Destination)
```
Comment (TomSweeneyRedHat) [inline libpod/container_internal_linux.go]: Would it be useful to also add the c.state.Mountpoint in the error message too?

**Comment by TomSweeneyRedHat** [comment]
One nit for consideration, otherwise LGTM if you can tame the tests.  The issues there don't appear related.

**Comment by mheon** [comment]
Bind mounts don't appear to use the same directory creation logic as tmpfs
mounts, at least in runc, so they don't seem to have issues. I could add
them easily enough, but I was a bit concerned with performance b
implications - we usually only have 3 or so tmpfs mounts, but a lot more
bind mounts

**Comment by rhatdan** [comment]
@mheon Ok fix @TomSweeneyRedHat Comment and then we can merge.

File: libpod/container_internal_linux.go (hunk 1 above)
Comment (mheon) [reply libpod/container_internal_linux.go]: Probably not - we don't really care where the container is mounted, specifically, only paths within the container's root filesystem

**Comment by mheon** [comment]
Tests green, should be good to merge

**Pull request merged**